### PR TITLE
Use Bazel for release builds / update_pods.py

### PR DESCRIPTION
**Pull request by jerrymarino**
This patch adds the ability to build with Bazel. We much prefer Bazel
due to its stability, reproducibility, and nix friendliness.

### CODE REVIEW

File: BUILD [hunk 1]
```diff
@@ -1,3 +1,53 @@
-# This is a blank file. In Bazel, it is a convention and requirement to have a
-# blank file if you want to reference some path.
-# The reason this exists, is because this directory is referenced by workspaces
+load("@build_bazel_rules_apple//apple:swift.bzl", "swift_library")
+load("@build_bazel_rules_apple//apple:macos.bzl", "macos_command_line_application")
+
+objc_library(
+    name = "ObjcSupport",
+    srcs = glob(["Sources/ObjcSupport/*.m"]),
+    hdrs = glob(["Sources/ObjcSupport/include/*"]),
```
Comment (jerrymarino) [inline BUILD]: This is underhanded, but it is how I am passing the custom module map.

File: WORKSPACE [hunk 2]
```diff
@@ -0,0 +1,12 @@
+git_repository(
+    name = "build_bazel_rules_apple",
+    remote = "https://github.com/jerrymarino/rules_apple.git",
+    tag = "0.4.0.2-pinterest",
```
Comment (rahul-malik) [inline WORKSPACE]: Latest rules apple has our patch so we can just point to 0.5.0

File: BUILD [hunk 3]
```diff
@@ -1,3 +1,53 @@
...
+)
+
+swift_library(
+    name = "RepoToolsCore",
+    srcs = glob(["Sources/RepoToolsCore/*.swift"]),
+    deps = [":PodToBUILD"],
+    copts = ["-swift-version", "4", "-static-stdlib"],
+)
+
+# TODO: (jerry) Setup test suite
```
Comment (rahul-malik) [inline BUILD]: we might as well do this in this PR if it isn't too much work

File: Makefile [hunk 4]
```diff
@@ -1,31 +1,34 @@
...
+release-spm: build-impl
+release-spm:
 	ditto .build/$(CONFIG)/Compiler bin/Compiler
 	ditto .build/$(CONFIG)/RepoTools bin/RepoTools
 
-build-impl:
+
+build-impl-spm:
 	swift build $(SWIFTBFLAGS) \
 	    -Xswiftc -target -Xswiftc x86_64-apple-macosx10.13
```
Comment (rahul-malik) [inline Makefile]: Why are we being explicit about the architecture here?

File: tools/bazelwrapper [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+set -e
+
+# Make sure we're in the project root directory.
+SCRIPTPATH="$( cd "$( dirname "${BASH_SOURCE[0]}"  )" && pwd  )"
+pushd "$SCRIPTPATH/.." > /dev/null
+trap popd > /dev/null ERR EXIT
+
+BAZEL_VERSION="0.12.0rc1"
+BAZEL_VERSION_SHA="fe9bf6e4d920c668f28a37d5d2ea11cee38871e90ccf5c2a52c7a327dd6de9a9"
+BAZEL_VERSION_URL="https://releases.bazel.build/0.12.0/rc1/bazel-0.12.0rc1-installer-darwin-x86_64.sh"
```
Comment (rahul-malik) [inline tools/bazelwrapper]: Want to point to the latest release? 0.13.0

File: Makefile (hunk 4 above)
Comment (jerrymarino) [reply Makefile]: How else would you set the target OS in SPM?

File: BUILD (hunk 3 above)
Comment (jerrymarino) [reply BUILD]: This needs:
- test running setup
- trans SwiftCheck things https://github.com/typelift/SwiftCheck/blob/master/Package.swift

I don't see much value in setting this up ATM, other than consistency

File: tools/bazelwrapper (hunk 5 above)
Comment (jerrymarino) [reply tools/bazelwrapper]: I've got a handful of tools building with and using the same version of Bazel - it's nice to atomically update them all at once. Perhaps this script can become it's own repo 👍

File: WORKSPACE (hunk 2 above)
Comment (jerrymarino) [reply WORKSPACE]: Done

**Comment by jerrymarino** [comment]
@rahul-malik merging this for now - the SwiftPM build blows up inside of `nix-shell`, while the Bazel works as expected.

Going forward, I'd like to test out Bazel for dev and test builds - we can do this as a follow up.

**Pull request merged**